operator/src/main/java/org/bf2/operator/managers/SecuritySecretManager.java

```diff
@@ -288,15 +289,14 @@ private Secret ssoClientSecretFrom(ManagedKafka managedKafka, Secret current) {
     }
 
     private Secret ssoTlsSecretFrom(ManagedKafka managedKafka, Secret current) {
-        if (!isKafkaAuthenticationEnabled(managedKafka)) {
-            return null;
-        }
-
-        return buildSecretFrom(ssoTlsSecretName(managedKafka),
-                "Opaque",
-                managedKafka,
-                current,
-                Map.of("keycloak.crt", managedKafka.getSpec().getOauth().getTlsTrustedCertificate()));
+        return Optional.ofNullable(managedKafka.getSpec().getOauth())
+            .map(ManagedKafkaAuthenticationOAuth::getTlsTrustedCertificate)
```

Comment:
But if we are changing the structure of this secret, don't we also have to change it in places where we're extracting this the TlsCert/key?

Reply:
it is not changing the structure of the secret, it is same. it inlines the isKafkaAuthenticationEnabled call in that first line.

Reply:
Okay, I understand now how it's the same output. Thanks!